test/test_view/test_project_publish.py

```diff
@@ -187,3 +187,53 @@ def test_unpublish_project(self):
         assert project.published, 'Project should be published'
         taskruns_when_published = cached_projects.n_task_runs(project.id)
         assert taskruns_when_published == orig_taskruns, 'Publish project should not have deleted taskruns'
+
+    @with_context
+    def test_newtask_unpublish_project(self):
+        """Test user obtains newtask with published projects only; 404 with unpublished projects"""
+        project = ProjectFactory.create(info=dict(published=True,
+            task_presenter='task presenter'))
+        make_subadmin(project.owner)
+        admin, regular_coowner, user = UserFactory.create_batch(3)
+        make_admin(admin)
+        tasks = TaskFactory.create_batch(10, project=project, n_answers=1)
+
+
+        self.set_proj_passwd_cookie(project, user)
+        res = self.app.get('api/project/{}/newtask?api_key={}'
+                           .format(project.id, user.api_key))
+        task = json.loads(res.data)
+        assert res.status_code == 200 and task['id'] == tasks[0].id, 'User should have obtained new task'
+
+        # submit answer for the task
+        task_run = dict(project_id=project.id, task_id=task['id'], info='hi there!')
+        res = self.app.post('api/taskrun?api_key={}'.format(user.api_key),
+                            data=json.dumps(task_run))
+        assert res.status_code == 200, res.status_code
+
+        #unpublish project
+        project.published = False
+        project.owners_ids.append(regular_coowner.id)
+        project_repo.save(project)
+        project = project_repo.get(project.id)
+        assert not project.published, 'Project should not be published'
+
+        # for unpublished project, obtaining newtask would succeed for admin
+        res = self.app.get('api/project/{}/newtask?api_key={}'
+                           .format(project.id, admin.api_key))
+        assert res.status_code == 200, 'newtask to unpublished project should succeed for admin'
+
+        # for unpublished project, obtaining newtask would succeed for subadmin
+        res = self.app.get('api/project/{}/newtask?api_key={}'
+                           .format(project.id, project.owner.api_key))
+        assert res.status_code == 200, 'newtask to unpublished project should succeed for admin'
```

Comment:
to be very nitpicking it should be project owner instead of admin, similar  below